sql/core/src/test/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListenerSuite.scala

```diff
@@ -43,13 +44,25 @@ import org.apache.spark.util.{AccumulatorMetadata, JsonProtocol, LongAccumulator
 import org.apache.spark.util.kvstore.InMemoryStore
 
 
-class SQLAppStatusListenerSuite extends SparkFunSuite with SharedSQLContext with JsonTestUtils {
+class SQLAppStatusListenerSuite extends SparkFunSuite with SharedSQLContext with JsonTestUtils
+  with BeforeAndAfter {
+
   import testImplicits._
 
   override protected def sparkConf = {
     super.sparkConf.set(LIVE_ENTITY_UPDATE_PERIOD, 0L).set(ASYNC_TRACKING_ENABLED, false)
   }
 
+  private var kvstore: ElementTrackingStore = _
+
+  before {
+    kvstore = new ElementTrackingStore(new InMemoryStore, sparkContext.conf)
```

Comment:
This instantiates this field for every test, even when it's not used. Instead, can you just check if it's null in 'after' and close/null it if so?

I suppose AppStatusListenerSuite does the same thing.

It's not a big deal I guess either way.

Reply:
Thanks @srowen for the comment.
Out of 10 test cases, 7 uses the kvstore.  
kvstore won't be null in 'after' block in any test cases. I think AppStatusListenerSuite doesn't check for null.

Reply:
@shahidki31 the above comment meant to ask you for setting the `kvstore` member val only in those tests where the store is really needed and then the null checks are needed to know when to close and release it.

Reply:
Thanks @attilapiros , I have updated.